calico_upgrade/pkg/migrate/migrate.go

```diff
@@ -911,6 +944,10 @@ func applyToBackend(clientv3 clientv3.Interface, kvp *model.KVPair) error {
 			if _, ok := err.(cerrors.ErrorResourceUpdateConflict); !ok {
 				break
 			}
+
+			// We hit a resource update conflict - pause for a short duration before
+			// retrying.
+			time.Sleep(time.Duration(float64(retryInterval) * (1 + (0.1 * rand.Float64()))))
 		}
 	}
 
```

Comment:
re: line 954, seems like we can be specific about whether we tried the update or not based on whether the error is `ErrorResourceAlreadyExists`. If it is, we know it was the update that failed, if its something else, we can report the create failed.